components/server/src/jobs/runner.ts

```diff
@@ -84,6 +86,7 @@ export class JobRunner {
                     reportJobCompleted(job.name, false);
                 } finally {
                     jobsDurationSeconds.observe(timer());
+                    await timeout;
```

Comment:
That's a clever way to ensure we wait at least as long as timeout.

Reply:
So the early release was actually by design, which ensures that whatever is locked runs **at least** once in that specified interval, but possibly more often. This change undoes that.

To actually fix the problem, it should've been enough to run the GC once an hour (as changed). This await shouldn't be here.

This now causes the StartWorkspace request, which uses this mutex, to have to wait 2 seconds before it can proceed, even if it's done earlier.

Reply:
> This now causes the StartWorkspace request, which uses this mutex, to have to wait 2 seconds before it can proceed, even if it's done earlier.

My bad, this is not the case actually as this only affects the jobs.

Reply:
Spoke on a call, we'll go with this. I think we could simplify and not use the `using` API (which is designed to hold the lock for the duration of the job) and instead just set an entry with a TTL, which we don't expire once the job is done.